.github/workflows/build-rapids.yml

```diff
@@ -0,0 +1,111 @@
+name: Build all RAPIDS repositories
+
+on:
+  workflow_call:
+
+jobs:
+  check-event:
+    name: Check GH Event
+    runs-on: ubuntu-latest
+    outputs:
+      ok: ${{ steps.check_gh_event.outputs.ok }}
+    steps:
+      - id: check_gh_event
+        name: Check GH Event
+        shell: bash
+        run: |
+          [[ '${{ github.event_name }}' == 'push' && '${{ github.repository }}' == 'NVIDIA/cccl' ]] || \
+          [[ '${{ github.event_name }}' == 'schedule' && '${{ github.repository }}' == 'NVIDIA/cccl' ]] || \
+          [[ '${{ github.event_name }}' == 'pull_request' && '${{ github.repository }}' != 'NVIDIA/cccl' ]] \
+          && echo "ok=true"  | tee -a $GITHUB_OUTPUT \
+          || echo "ok=false" | tee -a $GITHUB_OUTPUT;
+
+  build-rapids:
+    name: Build RAPIDS (${{ matrix.libs }})
+    if: needs.check-event.outputs.ok == 'true'
+    needs: check-event
+    secrets: inherit
+    # TODO: this is temporary
```

Comment:
Is this TODO still relevant?

Reply:
That depends. It's relevant if we want to port over some of the changes I made in CCCL's vendored copy of `build-in-devcontainer.yml` to the `rapidsai/shared-workflows` version. It's not relevant if we don't port them over.

I do think the changes made here are improvements and worth porting over, but I don't want to make that a blocker for merging this PR, which is why this file was vendored.